### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by sonokoh**
- enhanced add/edit features
- added drag and drop for sorting tasks
- added time support

### CODE REVIEW

File: ui/src/components/todo-list/QTodoItemEditDialog.js [hunk 1]
```diff
@@ -1,73 +1,64 @@
...
+                cancel: () => {
+                  this.$_cancel()
+                }
+              }
+            })
+          ])
+        ])
+      ]
     }
   },
```
Comment (cjshay) [inline ui/src/components/todo-list/QTodoItemEditDialog.js:50]: This function is a bit hard to read, maybe if you defined some of these as variables and passed them in that way? It's currently triple nested function-wise and hard to visually see where functions are starting / ending

This formatting isn't much better but easier for me to read personally (attached)
[todo-list-2.zip](https://github.com/Source-Digital/quasar-app-extension-tasks/files/5644416/todo-list-2.zip)

**Review by cjshay — COMMENTED** [review]
Low priority / Up to your discretion
1. Are you in control of the quickness of the transition between hour selection / minutes selection? It's a bit quick / jarring.
2. I really like the animation / functionality to remove the date by clicking it ==> unselecting it. What do you think about also adding a remove date / x button, for the person who might not realize clicking on the day on the calendar removes the date?
3. In http://localhost:8080/demo_-q-todo-list-edit, the edit needs some padding or a header in between the top of the input and the top of the edit modal (attached)
Screen Shot 2020-12-04 at 12 53 52 PM
4. How set in stone is this url routing schema? having a "_" and then a "-" seems a bit odd ex: demo_-q-todo-list-edit

**Comment by sonokoh** [comment]
Hi thanks for your feedbacks! 
> 1. Are you in control of the quickness of the transition between hour selection / minutes selection? It's a bit quick / jarring.
Unfortunately not, I'm using Quasar's QTime component as is. https://quasar.dev/vue-components/time#QTime-API

> 2. I really like the animation / functionality to remove the date by clicking it ==> unselecting it. What do you think about also adding a remove date / x button, for the person who might not realize clicking on the day on the calendar removes the date?
Great feedback! I added clear button.

> 3. In http://localhost:8080/demo_-q-todo-list-edit, the edit needs some padding or a header in between the top of the input and the top of the edit modal (attached)
> 4. How set in stone is this url routing schema? having a "_" and then a "-" seems a bit odd ex: demo_-q-todo-list-edit
These demo pages are development only, and will not be part of published modules so we don't have to worry too much -- but it was bugging me as well so I fixed it:) 
These URIs are auto generated by Quasar's predefined routing mechanism for dev mode. It automatically detects page components in ui/dev/src/pages and generates slug uri from their names.

File: ui/src/components/todo-list/QTodoItemEditDialog.js (hunk 1 above)
Comment (sonokoh) [reply ui/src/components/todo-list/QTodoItemEditDialog.js]: Yeah I agree that render function is hard to read in general.  I personally still prefer formatting below: 
```
h(Component, {
  props: {}
},[
    child1,
    child2
])
```

as the above gets rendered as:

```
<Component v-on="props">
  <child1 />
  <child2 />
</Component>
```
I created local variable so there's less complex nesting. I'll be careful going forward. Ty for the feedback:)

**Review by cjshay — APPROVED** [review]

**Pull request merged**